src/FactorGraph.jl

```diff
@@ -397,18 +401,32 @@ addVariable!(fg, :x0, Pose2)
 """
 function addVariable!(dfg::AbstractDFG,
                       label::Symbol,
-                      varType::InferenceVariable;
+                      varTypeU::Union{T, Type{T}}; 
```

Comment:
Cool thanks, I created a new const called `InstanceType{T}`:
https://github.com/JuliaRobotics/IncrementalInference.jl/blob/05d5d732fff020c5c4b7aad7b0813735bccc13a9/src/IncrementalInference.jl#L72

Reply:
I saw that one. I didn't think it would work as T kan be of type `InferenceVariable` and then you would get a method error in the softtype constructor and not in `addVarialbe!`. (in `varType()`)